nuxeo-java-client/src/main/java/org/nuxeo/client/api/objects/Document.java

```diff
@@ -258,6 +267,9 @@ public String getName() {
     }
 
     public void set(String key, Object value) {
+        if (value instanceof Calendar || value instanceof Date) {
+            throw new IllegalArgumentException(MSG_DATE_UNSUPPORTED);
+        }
```

Comment:
Shouldn't you also throw an exception if you use a _Calendar_ in the _setPropertyValue_ method ?